pubsub/rabbitmq/metadata.yaml

```diff
@@ -0,0 +1,168 @@
+      letter enabled). If both maxLen and maxLenBytes are set then both will 
+      apply; whichever limit is hit first will be enforced.
+    example: '1048576'
+  - name: concurrency
+    type: string
+    description: |
+      Allows processing multiple messages in
+      parallel (limited by the app-max-concurrency annotation, if configured).
+      Set to single to disable parallel processing. In most situations there’s 
+      no reason to change this.
+    example: '"parallel", "single"'
+    default: '"parallel"'
+  - name: enableDeadLetter
+    type: bool
+    description: |
+      Enable forwarding messages that cannot be handled to a dead-letter
+      topic.
+    default: '"false"'
+    example: '"true", "false"'
+  - name: prefetchCount
+    type: number
+    description: |
+      Number of messages to prefetch. Consider changing this to a non-zero
+      value for production environments. The value of "0" means that
+      all available messages will be pre-fetched.
+    default: '0'
+    example: '2'
+  - name: exchangeKind
+    type: string
+    description: |
+      Exchange kind of the rabbitmq exchange.
+    default: '"fanout"'
+    example: '"fanout","topic"'
+  - name: deliveryMode
+    type: number
+    description: |
+      Persistence mode when publishing messages.
+      RabbitMQ treats "2" as persistent, all other numbers as non-persistent.
+    default: '0'
+    example: '2'
+  - name: autoAck
+    type: bool
+    description: |
+      Whether or not the queue consumer should auto-ack messages.
+    default: '"false"'
+    example: '"true", "false"'
+  - name: deletedWhenUnused
+    type: bool
+    description: |
+      Whether or not the queue should be configured to auto-delete.
+    default: '"true"'
+    example: '"true", "false"'
+  - name: requeueInFailure
+    description: |
+      Whether or not to requeue when sending a negative acknowledgement
+      in case of a failure.
+    type: bool
+    default: '"false"'
+    example: '"true", "false"'
+  - name: reconnectWaitSeconds
+    description: |
+      Reconnect wait in Seconds.
+    type: duration
+  - name: ttlInSeconds
+    description: |
+      Default Queue TTL.
+    type: duration
+  - name: protocol
+    description: |
+      The RabbitMQ protocol. If the connectionString is specified,
+      this value will be set from URI scheme of the connectionString.
+      If both the connectionString and this property are specified,
+      the value specified here must match URI scheme of the connectionString.
+    type: string
+    example: '"amqp", "amqps"'
+  - name: caCert
+    type: string
+    description: |
+      Certificate authority certificate, required for using TLS.
+    example: "-----BEGIN CERTIFICATE-----\n<base64-encoded DER>\n-----END CERTIFICATE-----"
+  - name: clientCert
+    type: string
+    sensitive: true
+    description: |
+      Client certificate, required for authType mtls.
```

Comment:
I don't see mentions of "authType" set to "mtlS" anywhere in the component? And if it is an option, it should be a new auth profile.